src/main/java/seedu/address/model/person/Name.java

```diff
@@ -10,13 +10,23 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumeric characters, spaces, and the punctuations within the " +
+                    "list of allowed punctuations. " +
+                    "Names must start with an alphanumeric character.";
 
     /*
-     * The first character of the address must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
+     * Regex to check for valid punctuation
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String AllOWED_PUNCTUATION = "[\\?\\.\\'\\\"\\[\\]\\{\\}\\+\\^\\$\\*\\(\\)\\-<>,:;~@!#%&_=`]";
```

Comment:
Will update accordingly as per discussion during our meeting.

But generally, what the updated `VALIDATION _REGEX` does is that it enforces an alphanumeric character followed by any number of alphanumeric char, spaces, or the allowed punctuations.

The regex looks complicated because of the regex for `ALLOWED_PUNCTUATION`. The reason is because the punctuation we want to allow is a subset of the available punctuations. Creating a new regex class here resulted in a long and unreadable regex because of all the extra backslashes to escape relevant punctuations.